test/common/snap.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+const common = require('./');
+const path = require('node:path');
+const fs = require('node:fs/promises');
+const assert = require('node:assert/strict');
+
+
+const stackFramesRegexp = /(\s+)((.+?)\s+\()?(?:\(?(.+?):(\d+)(?::(\d+))?)\)?(\s+\{)?(\n|$)/g;
+
+function replaceStackTrace(str) {
+  return str.replace(stackFramesRegexp, '$1*$7\n');
+}
+
+function transform(...args) {
+  return (str) => args.reduce((acc, fn) => fn(acc), str);
+}
+
+function getSnapshotPath(filename) {
+  const { name, dir } = path.parse(filename);
+  return path.resolve(dir, `${name}.snapshot`);
+}
+
+async function assertSnapshot(actual, filename = process.argv[1]) {
+  const snapshot = getSnapshotPath(filename);
+  const expected = process.env.NODE_REGENERATE_SNAPSHOTS ? actual : await fs.readFile(snapshot, 'utf8');
+  if (process.env.NODE_REGENERATE_SNAPSHOTS) {
+    await fs.writeFile(snapshot, actual);
+  }
+  assert.strictEqual(actual, expected);
+}
+
+async function spawnAndAssert(filename, transform = (x) => x) {
+  const flags = common.parseTestFlags(filename);
+  const { stderr, stdout } = await common.spawnPromisified(process.execPath, [...flags, filename]);
+  await assertSnapshot(transform(`${stdout}${stderr}`), filename);
```

Comment:
Doesn't it matter what parts of the output came from `stdout` and which came from `stderr`?

Reply:
I think only in very specific (rare) cases it will make a difference

Reply:
I don’t think any of the current `message` tests distinguish between stdout and stderr?

For a test that needs to do so, it could be written using the `spawnPromisified` style and make its assertions explicitly. Ditto for all the `spawnPromisified` tests that check exit codes. I think it makes sense for the snapshot approach to stay lean and cover the core use case and leave `spawnPromisified` and its cousins for the edge cases.

Reply:
current behavior only compares stdout.
I don't think this is very important for this iteration since in all instances stdout is expected to be `''`

Reply:
> in all instances stdout is expected to be `''`

Do you mean stderr is expected to be empty? (Rather than stdout.)

Reply:
yep, my bad

Reply:
If we want to be able to quickly migrate lots of `message` tests to this new pattern, it might be preferable to match the current `message` comparison; so _just_ compare against `stdout`, and put a comment here that if the test author wants to compare against `stderr` instead or in addition, to use `spawnPromisified` instead. Because there might be `message` tests that would pass in the current Python comparison but fail in this, because `stderr` has an experimental warning or other miscellany that we don’t care about. Then we can migrate the `message` tests quickly without much refactoring.